File: utils/MMap.java
package wc.c.libbase.utils;


import java.util.HashMap;
import java.util.Map;

public final class MMap<KT, VT> {
    private Map<KT, VT> _this = null;

    public  MMap put(KT k, VT v) {
        _this.put(k, v);
        return this;
    }

    public VT get(KT k) {
        return _this.get(k);
    }

    public MMap() {
        _this = new HashMap<>();
    }

    public Map<KT, VT> get(){
        return _this;
    }
}

File: 7_项目搭建/android/AS/PositionCameraa/app/src/main/java/cyan/positioncameraa/utils/MMap.kt
package cyan.positioncameraa.utils

class MMap {
    //    private val _this: MutableMap<Any, Any> = mutableMapOf()
    private val _this: java.util.HashMap<String, Any> = HashMap()

    fun put(k: String, v: Any): MMap {
        _this.put(k, v)
        return this
    }

    fun <T> get(k: String, clz: Class<T>): T? {
        try {
            val o = _this.get(k)
            return clz.cast(o)
        } catch (e: Exception) {
            return null
        }
    }

    fun get(): HashMap<String, Any> {
        return _this
    }
}

